src/riak_cs_storage_mr.erl

```diff
@@ -0,0 +1,272 @@
+%% ---------------------------------------------------------------------
+%%
+%% Copyright (c) 2007-2013 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% ---------------------------------------------------------------------
+
+%% @doc MapReduce functions for storage calculation
+
+-module(riak_cs_storage_mr).
+
+-include("riak_cs.hrl").
+
+-export([bucket_summary_map/3,
+         bucket_summary_reduce/2]).
+-export([object_size_map/3,
+         object_size_reduce/2]).
+
+-export([bytes_and_blocks/1]).
+
+-ifdef(TEST).
+-export([object_size/1,
+         count_multipart_parts/1]).
+-endif.
+
+%% Record for summary information, each 3-tuple represents object
+%% count, total bytes and total blocks.  Some of them are estimated
+%% value, maybe (over|under)-estimated.
+%% @see bytes_and_blocks/1 for details of calculation/estimation
+-record(sum,
+        {
+          %% User accessible objects, which includes active and
+          %% writing of multipart.
+          user = {0, 0, 0},
+          %% `active's but "deleted", i.e. other than user accessible
+          ac_de = {0, 0, 0},
+          %% MP writing (double count with `user')
+          wr_mp = {0, 0, 0},
+          %% non-MP writing, divided by <> leeway
+          wr_new = {0, 0, 0},
+          wr_old = {0, 0, 0},
+          %% pending_delete, divided by <> leeway
+          pd_new = {0, 0, 0},
+          pd_old = {0, 0, 0},
+          %% scheduled_delete, divided by <> leeway
+          sd_new = {0, 0, 0},
+          sd_old = {0, 0, 0}
```

Comment:
Just for code readability, I'd also prefer not using two-chars-acronym. Maybe codes get ugly?